app/src/main/java/com/keylesspalace/tusky/AccountListActivity.kt

```diff
@@ -0,0 +1,102 @@
+/* Copyright 2017 Andrew Dawson
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.MenuItem
+
+import com.keylesspalace.tusky.fragment.AccountListFragment
+
+import javax.inject.Inject
+
+import dagger.android.AndroidInjector
+import dagger.android.DispatchingAndroidInjector
+import dagger.android.support.HasSupportFragmentInjector
+import kotlinx.android.synthetic.main.toolbar_basic.*
+
+class AccountListActivity : BaseActivity(), HasSupportFragmentInjector {
+
+    @Inject
+    lateinit var dispatchingAndroidInjector: DispatchingAndroidInjector<Fragment>
+
+    enum class Type {
+        FOLLOWS,
+        FOLLOWERS,
+        BLOCKS,
+        MUTES,
+        FOLLOW_REQUESTS,
+        REBLOGGED,
+        FAVOURITED
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_account_list)
+
+        val type = intent.getSerializableExtra(EXTRA_TYPE) as Type
+        val id: String? = intent.getStringExtra(EXTRA_ID)
+
+        setSupportActionBar(toolbar)
+        supportActionBar?.apply {
+            when (type) {
+                AccountListActivity.Type.BLOCKS -> setTitle(R.string.title_blocks)
+                AccountListActivity.Type.MUTES -> setTitle(R.string.title_mutes)
+                AccountListActivity.Type.FOLLOW_REQUESTS -> setTitle(R.string.title_follow_requests)
+                AccountListActivity.Type.FOLLOWERS -> setTitle(R.string.title_followers)
+                AccountListActivity.Type.FOLLOWS -> setTitle(R.string.title_follows)
+                AccountListActivity.Type.REBLOGGED -> setTitle(R.string.title_reblogged_by)
+                AccountListActivity.Type.FAVOURITED -> setTitle(R.string.title_favourited_by)
+            }
+            setDisplayHomeAsUpEnabled(true)
+            setDisplayShowHomeEnabled(true)
+        }
+
+        supportFragmentManager
+                .beginTransaction()
+                .replace(R.id.fragment_container, AccountListFragment.newInstance(type, id))
```

Comment:
Is it okay to do this? I have a feeling it can lead to duplicate fragments or replacing fragments even when not needed because they're retained automatically

Reply:
We have been using the sam in Java, so I guess its fine

Reply:
I know that we've been doing this but I think we unnecessarily reload the list on rotation now.